compiler/src/dotty/tools/dotc/tastyreflect/QuotedOpsImpl.scala

```diff
@@ -15,16 +21,28 @@ trait QuotedOpsImpl extends scala.tasty.reflect.QuotedOps with CoreImpl {
   def TermToQuoteDeco(term: Term): TermToQuotedAPI = new TermToQuotedAPI {
 
     def seal[T: scala.quoted.Type](implicit ctx: Context): scala.quoted.Expr[T] = {
-      typecheck()
-      new scala.quoted.Exprs.TastyTreeExpr(term).asInstanceOf[scala.quoted.Expr[T]]
-    }
 
-    private def typecheck[T: scala.quoted.Type]()(implicit ctx: Context): Unit = {
-      val tpt = QuotedTypeDeco(implicitly[scala.quoted.Type[T]]).unseal
-      if (!(term.tpe <:< tpt.tpe)) {
+      val expectedType = QuotedTypeDeco(implicitly[scala.quoted.Type[T]]).unseal.tpe
+
+      def etaExpand(term: Term): Term = term.tpe.widen match {
+        case mtpe: MethodType =>
+          val closureResType = mtpe.resType match {
+            case t: MethodType => t.toFunctionType()
+            case t => t
+          }
```

Comment:
It seems this does handle param blocks > 2?

Reply:
It was working properly but did not fully convert to the lambda type. I changed `toFunctionType ` to make sure it never leaves `MethodType`s as result type.